src/cmd_line/commands/breakpoints.ts

```diff
@@ -0,0 +1,271 @@
+  return arg.file !== undefined && arg.line !== undefined;
+}
+function isDelBreakpointHere(arg: any): arg is DelBreakpointHere {
+  return arg === 'here';
+}
+
+class DeleteBreakpointCommand extends ExCommand {
+  public override isRepeatableWithDot: boolean = false;
+  private readonly delBreakpoint: DelBreakpoint;
+
+  constructor(delBreakpoint: DelBreakpoint) {
+    super();
+    this.delBreakpoint = delBreakpoint;
+  }
+
+  async execute(vimState: VimState): Promise<void> {
+    if (isDelBreakpointById(this.delBreakpoint)) {
+      return vscode.debug.removeBreakpoints(
+        vscode.debug.breakpoints.splice(this.delBreakpoint.id - 1, 1)
+      );
+    } else if (isDelAllBreakpoints(this.delBreakpoint)) {
+      return vscode.debug.removeBreakpoints(vscode.debug.breakpoints);
+    } else if (isDelBreakpointFile(this.delBreakpoint)) {
+      let reqUri: vscode.Uri;
+      if (this.delBreakpoint.file === '') {
+        reqUri = vimState.document.uri;
+      } else {
+        const workspaceFolder =
+          vscode.workspace.getWorkspaceFolder(vimState.document.uri)?.uri ?? vscode.Uri.file('./');
+        reqUri = vscode.Uri.joinPath(workspaceFolder, this.delBreakpoint.file);
+      }
+      const reqLine = this.delBreakpoint.line - 1;
+      const breakpoint = vscode.debug.breakpoints
+        .filter(isSourceBreakpoint)
+        .find(
+          (b) =>
+            b.location.uri.toString() === reqUri.toString() &&
+            b.location.range.start.line === reqLine
+        );
+      if (breakpoint) return vscode.debug.removeBreakpoints([breakpoint]);
+    } else if (isDelBreakpointFunction(this.delBreakpoint)) {
+      const functionName = this.delBreakpoint.function;
+      const breakpoint = vscode.debug.breakpoints
+        .filter(isFunctionBreakpoint)
+        .filter((b) => b.functionName === functionName);
+      if (breakpoint) return vscode.debug.removeBreakpoints(breakpoint);
+    } else if (isDelBreakpointHere(this.delBreakpoint)) {
+      const location = new vscode.Location(vimState.document.uri, vimState.cursorStartPosition);
+      const distFromLocationCharacter = (b: vscode.SourceBreakpoint) => Math.abs(b.location.range.start.character - location.range.start.character);
+
+      const breakpoint = vscode.debug.breakpoints
+        .filter(isSourceBreakpoint)
+        .filter(
+          (b) =>
+            b.location.uri.toString() === location.uri.toString() &&
+            b.location.range.start.line === location.range.start.line
+        ).sort((a, b) => distFromLocationCharacter(a) - distFromLocationCharacter(b))[0];
+      if (breakpoint) return vscode.debug.removeBreakpoints([breakpoint]);
+    }
+  }
+}
+
+/**
+ * List Breakpoints Command
+ */
+class ListBreakpointsCommand extends ExCommand {
+  public override isRepeatableWithDot: boolean = false;
+
+  async execute(vimState: VimState): Promise<void> {
+    type AnyBreakpoint = vscode.Breakpoint & vscode.SourceBreakpoint & vscode.FunctionBreakpoint;
```

Comment:
Isn't this type equivalent to `vscode.Breakpoint`?

Reply:
not exactly. but I made it clearer now